### ISSUE TITLE
Build fails when installing Hadolint through yay on Endeavour OS

### ISSUE THREAD

**Issue by lucaspierru-convelio** [hadolint/hadolint#1027]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Build and install Hadolint on Endeavour OS with `yay -S hadolint` command 

### Actual behavior

Build fails with this error log:

```
==> Starting build()...
Preparing to install GHC (tinfo6) to an isolated location.
This will not interfere with any system-level installation.
Already downloaded.                       
gcc: error: unrecognized command-line option '--target=x86_64-unknown-linux'                
configure: error: Failed to determine machine word size. Does your toolchain actually work?
Received ExitFailure 1 when running
Raw command: /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2.temp/ghc-9.0.2/configure --prefix=/home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2/
Run from: /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2.temp/ghc-9.0.2/

                   
Error: Error encountered while configuring GHC with
         /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2.temp/ghc-9.0.2/configure --prefix=/home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2/
         run in /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2.temp/ghc-9.0.2/
       
       The following directories may now contain files, but won't be used by stack:
         - /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2.temp/
         - /home/lucaspierru/.stack/programs/x86_64-linux/ghc-tinfo6-9.0.2/
       
       For more information consider rerunning with --verbose flag
       
Configuring GHC ...
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: hadolint-exit status 4
 -> Failed to install the following packages. Manual intervention is required:
hadolint - exit status 4
```

### Steps to reproduce the behavior

```bash
yay -S hadolint
```

**Additional environment details (OS, stack version, etc.)**
Updated Endeavour OS with 
KDE Plasma 6.1.0
KDE Frameworks version: 6.3.0
QT version: 6.7.2
Kernel Version: 6.9.6-zen1-1-zen (64-bit)
Graphics Platform: Wayland

**Comment by Xalphor** [comment]
Same Here on Archlinux with kernel version `6.11.6-arch1-1`.

**Comment by Xalphor** [comment]
I fixed my problem by installing the pre-built `hadolint-bin` package.
